applicationset/controllers/applicationset_controller.go

```diff
@@ -1095,7 +1095,20 @@ func (r *ApplicationSetReconciler) updateApplicationSetApplicationStatus(ctx con
 
 		appOutdated := false
 		if progressiveSyncsRollingSyncStrategyEnabled(applicationSet) {
-			appOutdated = syncStatusString == "OutOfSync"
+			appOutdated = syncStatusString == string(argov1alpha1.SyncStatusCodeOutOfSync)
+			pruneEnabled := app.Spec.SyncPolicy != nil && app.Spec.SyncPolicy.Automated != nil && app.Spec.SyncPolicy.Automated.Prune
+			if !pruneEnabled {
+				requirePruneOnly := true
+				for _, r := range app.Status.Resources {
+					if r.Status != argov1alpha1.SyncStatusCodeSynced && !r.RequiresPruning {
+						requirePruneOnly = false
+						break
+					}
+				}
```

Comment:
@rumstead this would introduce another dependency on the in-CR resource status. How are you feeling about the Redis solution?

Reply:
thankfully progressive syncs only look at the top level application sync and health status. so we won't need any changes to the existing logic. the top level statuses are set by the application controller and aren't stored in redis.

Reply:
this change should also be fine, the `app.status.resources` are still populated when storing changes in redis, just the `app.status.resources[].health` would be missing. 

eg
```
status:
  health:
    status: Healthy
  resources:
  - group: apps
    health:
      status: Healthy
    kind: Deployment
    name: demo-app-kust
    namespace: cluster-generator
    status: OutOfSync
    version: v1
```
vs
```
status:
  health:
    status: Healthy
  resourceHealthSource: appTree
  resources:
  - group: apps
    kind: Deployment
    name: demo-app-kust
    namespace: cluster-generator
    status: OutOfSync
    version: v1
```